pkg/ibmvpc/examples/sg_testing1_new_with_grouping.txt

```diff
@@ -0,0 +1,11 @@
+
+grouped output:
+147.235.219.206/32 => subnet2-ky : protocol: TCP  dst-ports: 22
+subnet1-ky => 142.0.0.0/8,143.0.0.0/8 : protocol: ICMP  
+subnet1-ky => 161.26.0.0/16 : protocol: UDP  
+subnet2-ky => 142.0.0.0/8 : protocol: ICMP  
+subnet2-ky => subnet1-ky : All Connections
+subnet2-ky => vsi3b-ky[10.240.30.4] : protocol: TCP  
+subnet3-ky => subnet1-ky : All Connections
+vsi3b-ky[10.240.30.4] => subnet2-ky : protocol: TCP  
+vsi3b-ky[10.240.30.4] => vsi3a-ky[10.240.30.5] : All Connections
```

Comment:
We have to decide if we want to make the grouping-by-subnet more explicit in the report.
E.g., instead of using just the subnet name as the source-entry, put something like "[All network interfaces in <subnet-name>]"
@kyorav , what do you think?

Reply:
Makes sense, we cannot assume that everybody names their subnets "subnet##" like I do...
"[All network interfaces in ]" seems a bit long though. How about 
```
[subnet]subnet2-ky => 142.0.0.0/8 : protocol: ICMP
```
I imagine in the future we will have additional qualifiers there, like `[vpc]` or `[instance]`

Reply:
I like the proposed format. If square brackets are confusing, maybe use round brackets.
```
(subnet)subnet2-ky => 142.0.0.0/8 : protocol: ICMP
```

Reply:
I think it might be a bit misleading output (in the grouping context) to have "subnet" as a source/destination element (since the connectivity of its network interfaces might be grouped due to SG rule for which its members happen to include a certain subnet's interfaces). Not sure if replacing with "all network interfaces of subnet..." may help..

Reply:
We also considered using "endpoints". E.g.,
```
[all subnet subnet2-ky endpoints] => 142.0.0.0/8 : protocol: ICMP
```

Reply:
current suggestions after discussion with @kyorav  :

1. Keep the `subnet` term, and at the header of the report should explain it is referring to current configuration (only current configured network interfaces per subnet).
2. Generate group names per grouped network interfaces (e.g. `group1`) and refer to those names in the report + add section that details the interfaces included per group name.
3. Replace `subnet` with comma separated list of interfaces.
@zivnevo what do you think?

Reply:
Suggestion 1.  can work, but it feels like a compromise. I prefer items in a report to be self-explanatory rather than be explained in an out-of-context text.

Suggestions 2. and 3. are not expressing the fact that all network interfaces in the subnet share the same connectivity. I believe this bit of information might be important to the users (it ensures there are no outliers in the subnet).

Is my suggestion above too long or inaccurate?

Reply:
It seems a bit long, but mostly does not solve the problem of misleading people to think it covers any endpoint that would be added to the subnet. The accurate (or less misleading) text would be
```
[all existing endpoints in subnet subnet2-ky] => ...
```
which is even longer and very awkward. 

I don't love any of the solutions, so I vote for either living with the possibility that people might make wrong assumptions, and use `(subnet)subnet2-ky =>` or live with the long and awkward `[all existing endpoints in subnet subnet2-ky] =>`

Reply:
"the problem of misleading people to think it covers any endpoint that would be added to the subnet" - good point. I totally missed it. Taking all these factors into account I now vote for what was previously numbered as option 3: "Replace subnet with comma separated list of interfaces.". No confusions here, and this is also more consistent with how we group external IPs.

However, we may now want to change the code so that we group together NIs together even if they do not cover the whole subnet (but never group together NIs that come from different subnets).

Reply:
ok, updated the code accordingly